Homework4/Homework4/WeeklyTaskService.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Homework4
+{
+    internal class WeeklyTaskService
+    {
+        private readonly List<WeeklyTask> tasks;
+
+        public WeeklyTaskService()
+        {
+            tasks = new();
+        }
+
+        public void AddNewTask()
+        { 
+            Console.WriteLine("Add task in format: {},{},{},{}");
+            var inputData = Console.ReadLine();
+            var task = ParseNewTask(inputData);
+            AddNewTask(task);
+        }
+
+        public void UpdateTask()
+        {
+            Console.WriteLine("Input number to update:");
+            var inputNumber = Console.ReadLine();
+            var taskNumber = int.Parse(inputNumber);
+            
+            Console.WriteLine("Input new task data:");
+            var inputTaskData = Console.ReadLine();
+            var task = ParseNewTask(inputTaskData);
+            tasks[taskNumber - 1] = task;
```

Comment:
a well named method would be better 
```suggestion
            UpdateTask(taskNumber - 1, task);
```